### ISSUE TITLE
8.5.6 to 9.0.0 - Pushover integration / Screenshot delivery broken

### ISSUE THREAD

**Issue by jostrasser** [grafana/grafana#50911]
**What happened**:
Upgraded from 8.5.6 to 9.0.0. All alerts are migrated to Unified Alerting automatically.
After many hours of troubleshooting and customizing of the grafana.ini I can confirm that Unified Alerting in combination with Pushover does not deliver screenshots of the alert.

**What you expected to happen**:
If a alert is firing, it should deliver also a screenshot.
Also the delivered informations are really unclear now. It should look simple and deliver all relevant informations and not strange "var, metrics, labels...". They are not removable. It is also not possible to customize to remove the "rule_uid" in the subject of the alarm after the migration. Check out my comparison pls.

Tested also to send the same alert via mail, is working fine (with screenshot of stats panel).

**How to reproduce it (as minimally and precisely as possible)**:
Upgrade to 9.0.0, integrate Pushover and try to firing an alert with an attached screenshot.

**Anything else we need to know?**:
Customized the grafana.ini as follows, working for mail alerting, not for Pushover:

```
#################################### Alerting ############################
[alerting]
# Disable alerting engine & UI features
enabled = false
# Makes it possible to turn off alert rule execution but alerting UI is visible
;execute_alerts = true

[unified_alerting]
# Enable the Unified Alerting sub-system and interface. When enabled we'll migrate all of your alert rules and notification channels to the new system. New alert rules will be created and your notification channels will be converted into an Alertmanager configuration. Previous data is preserved to enable backwards compatibility but new data is removed when switching. When this configuration section and flag are not defined, the state is defined at runtime. See the documentation for more details.
enabled = true

[unified_alerting.screenshots]
# Enable screenshots in notifications. This option requires a remote HTTP image rendering service. Please
# see [rendering] for further configuration options.
capture = true

# The maximum number of screenshots that can be taken at the same time. This option is different from
# concurrent_render_request_limit as max_concurrent_screenshots sets the number of concurrent screenshots
# that can be taken at the same time for all firing alerts where as concurrent_render_request_limit sets
# the total number of concurrent screenshots across all Grafana services.
max_concurrent_screenshots = 5

# Uploads screenshots to the local Grafana server or remote storage such as Azure, S3 and GCS. Please
# see [external_image_storage] for further configuration options. If this option is false then
# screenshots will be persisted to disk for up to temp_data_lifetime.
upload_external_image_storage = true


#################################### External image storage ##########################
[external_image_storage]
# Used for uploading images to public servers so they can be included in slack/email messages.
# you can choose between (s3, webdav, gcs, azure_blob, local)
provider = local
```

**Environment**:
- Grafana version: 9.0.0
- Data source type & version: influxdb 1.8.x
- OS Grafana is installed on: LXD / Ubuntu Focal
- User OS & Browser: Safari / macOS 12.4
- Others: Pushover integration to deliver alerts via push notification to iPhones

Pushover Example 8.5.x:
![IMG_8796](https://user-images.githubusercontent.com/14274103/173936660-20245b34-6b3b-4d50-b3ba-197cef6c5638.PNG)

Pushover Example 9.0.0:
![IMG_8797](https://user-images.githubusercontent.com/14274103/173936710-91eaf9f2-717e-4669-807d-56b1507c30ee.jpeg)

**Comment by jostrasser** [comment]
No changes with 9.0.1. Didn't find any workaround yet.

**Comment by grobinson-grafana** [comment]
Hello!

Images in Pushover notifications are not supported right but should be available in 9.1.

> Also, when configuring external_image_storage as local upload_external_image_storage has to be set to false.

This isn't correct I'm afraid. Grafana is a valid external image store if your Grafana server is accessible via the Internet. If your Grafana server isn't accessible via the Internet, or you would just rather serve static assets from an object store then you can choose between Amazon S3, Azure Blob Storage and Google Cloud Storage.

If you have set Grafana as your external image store such that `upload_external_image_storage` is true, and your Grafana server is accessible at `https://grafana.mydomain.com`, then you can expect to find `myimage.png` at `https://grafana.mydomain.com/public/img/attachments/myimage.png`.

**Comment by grobinson-grafana** [comment]
Hello!

Images in Pushover notifications are not supported right but should be available in 9.1.

> Also, when configuring external_image_storage as local upload_external_image_storage has to be set to false.

This isn't correct I'm afraid. Grafana is a valid external image store if your Grafana server is accessible via the Internet. If your Grafana server isn't accessible via the Internet, or you would just rather serve static assets from an object store then you can choose between Amazon S3, Azure Blob Storage and Google Cloud Storage.

If you have set Grafana as your external image store and `upload_external_image_storage` is true, and your Grafana server is accessible at https://grafana.mydomain.com, then you can expect to find `myimage.png` at https://grafana.mydomain.com/public/img/attachments/myimage.png.

If `upload_external_image_storage` is false then Grafana will attempt to upload the image from the local disk where supported, but at present we just support uploading for Discord and email.

**Comment by grobinson-grafana** [comment]
Hello!

Images in Pushover notifications are not supported at this time but should be available in 9.1.

> Also, when configuring external_image_storage as local upload_external_image_storage has to be set to false.

This isn't correct I'm afraid. Grafana is a valid external image store if your Grafana server is accessible via the Internet. If your Grafana server isn't accessible via the Internet, or you would just rather serve static assets from an object store then you can choose between Amazon S3, Azure Blob Storage and Google Cloud Storage.

If you have set Grafana as your external image store and `upload_external_image_storage` is true, and your Grafana server is accessible at https://grafana.mydomain.com, then you can expect to find `myimage.png` at https://grafana.mydomain.com/public/img/attachments/myimage.png.

If `upload_external_image_storage` is false then Grafana will attempt to upload the image from the local disk where supported, but at present we just support uploading for Discord and email.

**Comment by grobinson-grafana** [comment]
Hello!

Images in Pushover notifications are not supported at this time but should be available in 9.1.

> Also, when configuring external_image_storage as local upload_external_image_storage has to be set to false.

This isn't correct I'm afraid. Grafana is a valid external image store if your Grafana server is accessible via the Internet. If your Grafana server isn't accessible via the Internet, or you would just rather serve static assets from an object store then you can choose between Amazon S3, Azure Blob Storage and Google Cloud Storage.

If you have set Grafana as your external image store and `upload_external_image_storage` is true, and your Grafana server is accessible at https://grafana.mydomain.com, then you can expect to find `myimage.png` at https://grafana.mydomain.com/public/img/attachments/myimage.png.

If `upload_external_image_storage` is false then Grafana will attempt to upload the image from the local disk where supported, but at present we just support uploading for Discord and email.

**Comment by jostrasser** [comment]
Thanks for your detailed explanation @grobinson-grafana !

I have now set my config back to the original posted one:

`[unified_alerting]
upload_external_image_storage = true

[external_image_storage]
provider = local`


Works as expected I think. 

Good to know that Grafana 9.1 brings the support for images back to Pushover Notifications! 😃 

Thank you!

**Comment by grobinson-grafana** [comment]
Hi @jostrasser,

There is a small mistake in the configuration as `upload_external_image_storage` should be under `[unified_alerting.screenshots]`:

```
[unified_alerting.screenshots]
upload_external_image_storage = true

[external_image_storage]
provider = local
```

**Comment by grobinson-grafana** [comment]
Hi @jostrasser,

There is a small mistake in the configuration as `upload_external_image_storage` should be under `[unified_alerting.screenshots]`:

```
[unified_alerting.screenshots]
upload_external_image_storage = true

[external_image_storage]
provider = local
```

**Comment by grobinson-grafana** [comment]
Hi @jostrasser,

There is a small mistake in the configuration as `upload_external_image_storage` should be under `[unified_alerting.screenshots]`:

```
[unified_alerting.screenshots]
upload_external_image_storage = true

[external_image_storage]
provider = local
```

**Comment by grobinson-grafana** [comment]
Hi @jostrasser,

There is a small mistake in the configuration as `upload_external_image_storage` should be under `[unified_alerting.screenshots]`:

```
[unified_alerting.screenshots]
upload_external_image_storage = true

[external_image_storage]
provider = local
```

**Comment by grobinson-grafana** [comment]
Hi @jostrasser,

There is a small mistake in that configuration as `upload_external_image_storage` should be under `[unified_alerting.screenshots]`:

```
[unified_alerting.screenshots]
upload_external_image_storage = true

[external_image_storage]
provider = local
```

**Comment by grobinson-grafana** [comment]
Hi, a quick update! I expect Pushover to be available in 9.0.2 this week!

**Comment by jostrasser** [comment]
Thanks for the update @grobinson-grafana !
